include/swift/AST/DiagnosticsSema.def

```diff
@@ -4613,6 +4613,9 @@ ERROR(concurrent_access_of_inout_param,none,
 ERROR(non_sendable_capture,none,
       "capture of %1 with non-sendable type %0 in a `@Sendable` closure",
       (Type, DeclName))
+ERROR(implicit_non_sendable_capture,none,
+      "implicit capture of %1 requires that %0 conforms to `Sendable`",
```

Comment:
I think this explains compiler details but doesn't really help users, how about something like `cannot capture non-sendable type <decl-name> in 'async let' binding <binding-name>' or `'async let' binding <binding-name> cannot capture non-sendable <decl-name>' and a note to add `: Sendable` or something like that.

@hborla WDTY?

Reply:
Thanks @xedin, I'm making the adjustments. But initially I'm thinking in `capture of <decl-name> with non-sendable type <type> in 'async let' binding` to remain consistent with `non_sendable_capture` one. WDYT?

Reply:
Sounds good which a minor change - if you going to use `type` then print a type and point to the declaration with a note.